File: generators/python-generators/equivalent_trees.py
import random

class Tree(object):
    def __init__(self, left, value, right):
        self.value = value
        self.left = left
        self.right = right
    
    def insert(self, value):
        if value < self.value:
            return Tree(self.left.insert(value) if self.left else Tree(None, value, None), self.value, self.right)
        elif value > self.value:
            return Tree(self.left, self.value, self.right.insert(value) if self.right else Tree(None, value, None))
        else:
            return self
    
    def __repr__(self):
        return "Tree(%s, %s, %s)" % (self.left, self.value, self.right)


    def walk(self):
        pass # TODO: implement this

def new_tree(seed):
    random.seed(seed)
    tree = Tree(None, random.randrange(0, 100), None)
    for i in range(10):
      tree = tree.insert(random.randrange(0, 100))
    return tree


def same(t1, t2):
    pass # TODO: implement this

print(same(new_tree(0), new_tree(0)))
print(same(new_tree(0), new_tree(1)))

File: generators/js-generators/equivalent_trees.js
var seed = require('seed-random');

class Tree {
  constructor(left, value, right) {
    this.left = left;
    this.value = value;
    this.right = right;
  }

  static new_tree(s) {
    let rand = seed(s)
    let tree = new Tree(null, rand(), null);
    for (let i = 0; i < 10; i++) {
      tree = Tree.insert(tree, rand());
    }
    return tree;
  }

  static insert(tree, value) {
    if (tree === null) {
      return new Tree(null, value, null);
    } else if (value < tree.value) {
      return new Tree(Tree.insert(tree.left, value), tree.value, tree.right);
    } else {
      return new Tree(tree.left, tree.value, Tree.insert(tree.right, value));
    }
  }

  *walk() {
    // TODO: Implement this function
  }
}

function same(tree1, tree2) {
  // TODO: Implement this function
}

console.log(same(Tree.new_tree('0'), Tree.new_tree('0')))
console.log(same(Tree.new_tree('0'), Tree.new_tree('1')))